### ISSUE TITLE
Problems with Javascript (error $ not defined) / NPM install

### ISSUE THREAD

**Issue by lrmmf** [Microsoft/PowerBI-visuals#116]
Hi,
first of all, im not a web developer so i'm probably missing something in these 2 issues:
My 1st problem is this:
1.1) if i fork Microsoft PowerBI-Visuals, i can run the project in visual studio (though i had to execute npm clear cache or something like that before npm install). - OK
1.2) if i fork SQL-BI's Bullet Chart or KPI indicator with status, deviation and history... when i run npm install i sometimes got errors (happened in several attempts), but now, while not appearing any errors when NPM runs, in Visual Studio i always get errors. 

I would like to attach two files, 1) the Powershell screen  when NPM runs (seems ok), and the task runner explorer log (not ok), but i can't upload txt ou docx files!

My second problem is this:
2) if i download these specific directories and run then locally, i always got the javascript error $ is not defined! What i'm i missing? I don't see anyone speaking of this! i know that this is related to jquery and tried some things in the startup page índex.html in the playground directory:

The error happens here ...
   <script type="text/javascript">
        $(document).ready(function () { <-------------------------- in the $ sign.
            powerbi.visuals.Playground.initialize();
        });
    
Regards and thanks for your help.

**Comment by ebunt** [comment]
send a screenshot of your errors.

**Comment by lrmmf** [comment]
![2015-09-27_09h03_14](https://cloud.githubusercontent.com/assets/5270111/10121820/05b2abb8-64f7-11e5-9156-e121586b359d.png)
![2015-09-27_09h03_30](https://cloud.githubusercontent.com/assets/5270111/10121821/05b631d4-64f7-11e5-813f-4083cd79176f.png)
![2015-09-27_09h03_51](https://cloud.githubusercontent.com/assets/5270111/10121824/05d645f0-64f7-11e5-989a-8f6431009dc2.png)
![2015-09-27_09h04_07](https://cloud.githubusercontent.com/assets/5270111/10121827/05dacd32-64f7-11e5-8b72-5728c4e5f763.png)
![javascript error](https://cloud.githubusercontent.com/assets/5270111/10121826/05da7508-64f7-11e5-9e00-4c2b00700d06.png)
![2015-09-27_09h02_18](https://cloud.githubusercontent.com/assets/5270111/10121825/05d99408-64f7-11e5-963d-85a4ab16cd47.png)
![2015-09-27_09h02_42](https://cloud.githubusercontent.com/assets/5270111/10121823/05d2f54e-64f7-11e5-84fe-099d8c08566b.png)
![2015-09-27_09h02_59](https://cloud.githubusercontent.com/assets/5270111/10121822/05d2b052-64f7-11e5-8b64-3cc904eb5e31.png)

**Comment by lrmmf** [comment]
I send one printscreen of the $ error and 7 from the task runner explorer.
Regards.

**Comment by bspillar** [comment]
Any solution to this?

**Comment by bspillar** [comment]
I think since my NPM install does not complete successfully some of these .js files end up missing. I'd really like to get my solution set up, but I keep running into problems. I got it going last week and now I'm unsuccessful this week.

**Comment by bspillar** [comment]
A couple things that helped me: On my npm install I was getting errors. This was a second install for me which may be the reason for my errors. I ran the command 'npm cache clean' and then 'npm install' and was able to get a successful install. Once I had a successful install and opened the solution, in addition to the steps listed in the set up documents, I also had to do an initial build in order to have some of the required .js files created.
